### ISSUE TITLE
In Expanded view, auto-scroll to the root when clicking on the root

### ISSUE THREAD

**Issue by khakhalin** [Zettlr/Zettlr#813]
(for Windows version 1.6.0)

So this is a very simple behavior, but it may be a bit hard to describe, so the explanation may be a bit wordy. 

Imagine this use case: a person has a project with some subfolders, open in Expanded view. In this view, the leftmost panel contains the list of folders, while the mid-panel has the list of all notes. Now, imagine that some notes are stored in subfolders, but some are kept in the top-level (root) folder.

Currently there is no way to scroll to these "root folder" notes without actually using a mouse. It is not a problem for subfolders: if you click a subfolder in the left list, the right list scrolls to the contents of this subfolder. If you click on the root folder, however, it scrolls to the _lastest active file_. If you are lucky, and the latest active file was in the root, it will show the root. But more often than not, the latest active file was _not_ in the root, and the only way to scroll to the root is to use the mouse wheel. Which is painful. Or catch the slider, and move it upwards. Which is also painful. The "Home" button on the keyboard also doesn't work, as the 2nd panel is not in focus after you clicked on something in the first panel, so pressing "Home" doesn't help. (You can of course click on some random file in the 2nd panel, and then hit "Home" on the keyboards, but it is also painful).

I propose to change the behavior just a little bit, so that clicking on the root folder felt "equivalent" to clicking on subfolders. Clicking on a subfolder shows the first files in this subfolder, and so clicking on the root folder should ideally also show the first files of the root folder.

**Comment by nathanlesage** [comment]
> You can of course click on some random file in the 2nd panel, and then hit "Home" on the keyboards, but it is also painful

You don't know how relieved I am (seriously!) that pressing `Home` for Windows user actually scrolls up the file list :D 

> I propose to change the behavior just a little bit, so that clicking on the root folder felt "equivalent" to clicking on subfolders. Clicking on a subfolder shows the first files in this subfolder, and so clicking on the root folder should ideally also show the first files of the root folder.

So basically what you're asking for is "If I click on a root folder, albeit it's already selected, please scroll up all the way so that I am at the root of the folder again"?

**Comment by khakhalin** [comment]
Heh, I obviously did a bad job explaining :) Let me try with some printscreens. So here's a staring position: I'm working with some file from the root folder:

![image](https://user-images.githubusercontent.com/17154125/82134820-e8e89a00-97c9-11ea-8afe-6c0f2cd3a664.png)

Now I click on some subfolder. The 2nd list scrolls to its content, which is great:

![image](https://user-images.githubusercontent.com/17154125/82134826-ff8ef100-97c9-11ea-8b22-0098d0033c70.png)

If at this point if I immediately click back on my root folder (in this case `_Notes-git`), the 2nd list WILL rotate back. Because the active file didn't change. HOWEVER let's say I click on some file in the subfolder, and work with it for a while. (The same thing will happen if I navigated to this file via `[[ref]]` links. Now this file is active:

![image](https://user-images.githubusercontent.com/17154125/82134845-38c76100-97ca-11ea-9c93-bfd9e3cf85a2.png)

If I click on my root folder NOW, I won't return to the list of files in the root folder. What will happen is that the right list will scroll up as high as possible, to keep the currently-active file still in view. In practice it means that I'll get some view half-way between the subfolder I visited, and the root folder to which I hoped to return.

![image](https://user-images.githubusercontent.com/17154125/82134904-e470b100-97ca-11ea-88d1-6fce67ab2145.png)

This view is not useful at all, and from there I can only navigate upwards (back to the list of files in the root) in three ways. Either using the mouse wheel, or using the mouse+right slider, or by clicking to the 2nd window once again (to give it focus), and then hitting "Home on the keyboard". In practice I use the wheel, but it means so much wheeling during a typical session that my index finger starts to hurt.

Essentially I propose that in this mode, the folders in the leftmost list serve more like folders in a file system, and scroll to the top files in the subfolder, or the top folder, without trying to keep the currently-active-file visible in the list. If a user clicked on the folder, they are probably interested in the contents of this folder, and wish to interact with it; the immediate vicinity of the currently-active-note is no longer relevant (or at least less relevant).

Does it makes sense? Or am I in the minority on this one? :)

**Comment by nathanlesage** [comment]
it's super interesting to always read "Zettlr scrolls down", because it doesn't :D 

Every time you click on any folder, what is actually happening is that Zettlr displays you the contents of that folder in the file list, hence the topmost folder of the list is always the selected one.

Which brings me to the next point: While I can certainly see your point, the big, big problem is: How to decide when the file should have precedence (i.e. try to keep the active file as visible as possible) versus when the newly selected folder should have precedence (i.e. keep it at the top without scrolling the file into view)

**Comment by khakhalin** [comment]
I am sorry, I probably don't quite understand what you said here, but what's happening with my Zettlr is that it does NOT show the content of the root folder. Maybe it's a difference between operation systems? On my system, it scrolls correctly for all other folders, but fails to scroll all the way for the root one.

I'll try to make a video, as it seems that my descriptions still miss something (sorry!). But the tldr is that it scrolls to all folders but the root one.

Of course, one solution for the time being could be to not use the root folder at all, and keep all notes in subfolders. I may have to switch to this layout, as I've developed something like a nerve damage from scrolling, and it's starting to become a usability issue :)

**Comment by khakhalin** [comment]
Here, I made a youtube video with the use case. Hopefully this way it will be easier to relate :)

https://www.youtube.com/watch?v=l3nvn2F6k_c

**Comment by nathanlesage** [comment]
I think I understand now where you're stumbeling — it's a certain functionality the file list exhibits that I took over from Ulysses.

So what Zettlr is doing whenever you select a directory is that it takes the directory contents (all of them) and places them in the file list; that is, the file list always _starts_ with the directory you've selected. The inconsistency now occurs whenever you switch between different _levels_ of directories. If you have, say, two subdirectories in your root on the same level (e.g. level 2, directly beneath the root), and you select a file in any of them, then what happens when you select the other one is that the selected file obviously is not present in the other directory, so nothing happens. But as soon as you traverse back to the root directory, the file is obviously back in the file list, so Zettlr will attempt to scroll it back into view.

What I just thought of was maybe a compromise — I cannot say if this breaks anything, but what if the file list _always displays the complete root contents_, and clicking on a directory merely scrolls the directory into view? Then you could "go back" to the root directory without having to select it, and the directory selection would take precedence. This might not even break a single workflow. However, I am not sure about the caveats of this, but I can certainly see some benefits from this …

**Comment by khakhalin** [comment]
I'm adding some activity here, only to de-stale the thread, and prevent it from auto-closing :) A smoother navigation through the folder list would still be much appreciated!
